src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkUnusedLogFileInspection.kt

```diff
@@ -28,21 +27,22 @@ class SmkUnusedLogFileInspection : SnakemakeInspection() {
 
         override fun visitSmkUse(use: SmkUse) {
             val logSection = use.getSectionByName(SnakemakeNames.SECTION_LOG) ?: return
-            use.getOverriddenRuleReferences()?.forEach {
+            use.getImportedRuleNames()?.children?.forEach {
                 var resolveResult: PsiElement?
                 var reference = it
+                // Searching for origin element, it must be a rule or checkpoint
                 while (true) {
-                    resolveResult = reference.reference.resolve()
+                    resolveResult = reference.reference?.resolve()
                     when (resolveResult) {
-                        is SmkRule -> break
+                        is SmkRuleOrCheckpoint -> break
                         is SmkReferenceExpression -> reference = resolveResult
                         else -> {
                             return@forEach
                         }
                     }
                 }
                 if (resolveResult != null) {
-                    resolveResult as SmkRule
+                    resolveResult as SmkRuleOrCheckpoint
                     val ruleLogSection = resolveResult.getSectionByName(SnakemakeNames.SECTION_LOG)
                     val message =
```

Comment:
Local returns like `return@forEach` are hard to track in complex code. Please extract foreach method lambda into a separate method

Reply:
Additional method `handleUseReference` is created